Functional.Maybe/MaybeAsync.cs

```diff
@@ -30,6 +31,15 @@ public static async Task<T> OrElse<T>(this Task<Maybe<T>> @this, Func<T> orElse)
 		{
 			var res = await @this;
 			return res.HasValue ? res.Value : orElse();
+		} 
+		public static async Task<Maybe<T>> ToMaybeAsync<T>(this Task<T> task)
+		{
+			return (await task).ToMaybe();
+		}
+
+		public static async Task<Maybe<T>> ToMaybeAsync<T>(this Task<T?> task) where  T : struct
+		{
+			return (await task).ToMaybe();
 		}
-	}
+  }
```

Comment:
And this tab to whitespace conversion seems erroneous to me. Can you explain it, please?

Reply:
Yes, this is my mistake. I had my editor set to indent by spaces and when I noticed that the codebase uses tabs, I switched and tried to fix my mistakes. This one slipped my attention.

Reply:
Hopefully fixed.